Build failure because of a registered MOOSE object

## Bug Description
While updating Cardinal submodule in SALAMANDER https://github.com/idaholab/salamander/pull/133 the build fails because OpenMCCellTransformBase is registered as a Moose object but does not derive from MooseObject. error shown below:

```
In file included from /tmp/build/salamander/moose/framework/include/base/MooseObject.h:17,
                 from /tmp/build/salamander/cardinal/include/base/OpenMCBase.h:21,
                 from /tmp/build/salamander/cardinal/src/transforms/OpenMCCellTransformBase.C:25,
                 from /tmp/build/salamander/cardinal/build/unity_src/transforms_Unity.C:2:
/tmp/build/salamander/moose/framework/include/base/Registry.h: In instantiation of 'RegistryEntry<T>::RegistryEntry(const RegistryEntryData&) [with T = OpenMCCellTransformBase]':
/opt/rh/gcc-toolset-13/root/usr/include/c++/13/bits/stl_construct.h:119:7:   required from 'void std::_Construct(_Tp*, _Args&& ...) [with _Tp = RegistryEntry<OpenMCCellTransformBase>; _Args = {const RegistryEntryData&}]'
/opt/rh/gcc-toolset-13/root/usr/include/c++/13/bits/alloc_traits.h:661:19:   required from 'static void std::allocator_traits<std::allocator<void> >::construct(allocator_type&, _Up*, _Args&& ...) [with _Up = RegistryEntry<OpenMCCellTransformBase>; _Args = {const RegistryEntryData&}; allocator_type = std::allocator<void>]'
/opt/rh/gcc-toolset-13/root/usr/include/c++/13/bits/shared_ptr_base.h:604:39:   required from 'std::_Sp_counted_ptr_inplace<_Tp, _Alloc, _Lp>::_Sp_counted_ptr_inplace(_Alloc, _Args&& ...) [with _Args = {const RegistryEntryData&}; _Tp = RegistryEntry<OpenMCCellTransformBase>; _Alloc = std::allocator<void>; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]'
/opt/rh/gcc-toolset-13/root/usr/include/c++/13/bits/shared_ptr_base.h:971:16:   required from 'std::__shared_count<_Lp>::__shared_count(_Tp*&, std::_Sp_alloc_shared_tag<_Alloc>, _Args&& ...) [with _Tp = RegistryEntry<OpenMCCellTransformBase>; _Alloc = std::allocator<void>; _Args = {const RegistryEntryData&}; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]'
/opt/rh/gcc-toolset-13/root/usr/include/c++/13/bits/shared_ptr_base.h:1712:14:   required from 'std::__shared_ptr<_Tp, _Lp>::__shared_ptr(std::_Sp_alloc_shared_tag<_Tp>, _Args&& ...) [with _Alloc = std::allocator<void>; _Args = {const RegistryEntryData&}; _Tp = RegistryEntry<OpenMCCellTransformBase>; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]'
/opt/rh/gcc-toolset-13/root/usr/include/c++/13/bits/shared_ptr.h:464:59:   required from 'std::shared_ptr<_Tp>::shared_ptr(std::_Sp_alloc_shared_tag<_Tp>, _Args&& ...) [with _Alloc = std::allocator<void>; _Args = {const RegistryEntryData&}; _Tp = RegistryEntry<OpenMCCellTransformBase>]'
/opt/rh/gcc-toolset-13/root/usr/include/c++/13/bits/shared_ptr.h:1009:14:   required from 'std::shared_ptr<typename std::enable_if<(! std::is_array< <template-parameter-1-1> >::value), _Tp>::type> std::make_shared(_Args&& ...) [with _Tp = RegistryEntry<OpenMCCellTransformBase>; _Args = {const RegistryEntryData&}; typename enable_if<(! is_array< <template-parameter-1-1> >::value), _Tp>::type = RegistryEntry<OpenMCCellTransformBase>]'
/tmp/build/salamander/moose/framework/include/base/Registry.h:185:57:   required from 'static char Registry::add(const RegistryEntryData&) [with T = OpenMCCellTransformBase]'
/tmp/build/salamander/cardinal/src/transforms/OpenMCCellTransformBase.C:28:1:   required from here
/tmp/build/salamander/moose/framework/include/base/Registry.h:361:51: error: static assertion failed: Not derived from MooseObject or Action
  361 |   static_assert(std::is_base_of_v<MooseObject, T> || std::is_base_of_v<Action, T>,
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/build/salamander/moose/framework/include/base/Registry.h:361:51: note: '(((bool)std::is_base_of_v<MooseObject, OpenMCCellTransformBase>) || ((bool)std::is_base_of_v<Action, OpenMCCellTransformBase>))' evaluates to false
make: *** [/tmp/build/salamander/moose/framework/build.mk:166: /tmp/build/salamander/cardinal/build/unity_src/transforms_Unity.x86_64-pc-linux-gnu.opt.lo] Error 1
make: *** Waiting for unfinished jobs....
[07:39:42] ERROR: Exiting with code 2

[07:39:42] Execution in moosebuild completed with return code 2
[07:39:42] Submitting step statistics
[07:39:42] Failed to submit step statistics
```

## Steps to Reproduce
error shown on Civet - probably can reproduce on Cardinal when building with a newer MOOSE version

## Impact
stops the updates

--- a/src/transforms/OpenMCCellTransformBase.C
+++ b/src/transforms/OpenMCCellTransformBase.C
@@ -25,8 +25,6 @@
 #include "OpenMCBase.h"
 #include "UserErrorChecking.h"
 
-registerMooseObject("CardinalApp", OpenMCCellTransformBase);
-
 InputParameters
 OpenMCCellTransformBase::validParams()
 {

